ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -494,21 +490,33 @@ class Endpoints(
 
   private def aggregateListUserPages(
       token: Option[String],
-      pageToken: String = "",
       pageSize: Int = 1000, // TODO could be made configurable in the future
-  ): Future[Seq[User]] =
-    userManagementClient.listUsers(token, pageToken, pageSize).flatMap {
-      case (users, "") => Future.successful(users)
-      case (users, pageToken) => aggregateListUserPages(token, pageToken, pageSize).map(users ++ _)
+  ): Source[Error \/ User, NotUsed] = {
```

Comment:
I see, so this is how we create a chunked response then. Very nice!

Reply:
The ability to do this was one of the motivating factors behind our initial impulse to always return 200 HTTP responses and require clients to check the in-body status code instead. Because we can't detect a late error and update our HTTP response, but because we can arrange the JSON encoding such that status is at the end, we can detect the late error and report the failed status that way.

We ended up reporting HTTP status anyway, so the interface is kind of inconsistent. I hope this isn't too much of a problem in practice.